### PR TITLE
feat: support Custom Reference Modules

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/domain/src/use-cases/publications/CreateQuote.ts [hunk 1]
```diff
@@ -52,11 +34,7 @@ export class CreateQuote extends SponsorshipReady<CreateQuoteRequest> {
   }
 
   protected override async sponsored(request: CreateQuoteRequest): Promise<void> {
-    if (
-      request.reference.type === ReferencePolicyType.ANYONE &&
-      request.actions.length === 0 &&
-      isMomokaPublicationId(request.quoteOn)
-    ) {
```
Comment (cesarenaldi) [inline packages/domain/src/use-cases/publications/CreateQuote.ts:59]: I don't know what I was thinking, just the `quoteOn` is relevant for making a decision on Momoka or on-chain. Same applies to mirrors and comments.

File: packages/react/src/transactions/adapters/publications/CreateOnChainMirrorGateway.ts [hunk 2]
```diff
@@ -144,6 +145,8 @@ export class CreateOnChainMirrorGateway
   private resolveOnchainMirrorRequest(request: CreateMirrorRequest): OnchainMirrorRequest {
     return {
       mirrorOn: request.mirrorOn,
+      mirrorReferenceModuleData: request.mirrorOnReferenceData,
```
Comment (krzysu) [inline packages/react/src/transactions/adapters/publications/CreateOnChainMirrorGateway.ts:148]:
```suggestion
      mirrorOnReferenceModuleData: request.mirrorOnReferenceData,
```
?

File: packages/react/src/transactions/publications/useExecutionMode.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  const { execute: fetchModuleMetadata } = useLazyModuleMetadata();
+
+  return async ({
+    actions,
+    author,
+    reference,
+    referencedPublicationId,
+    sponsored,
+  }: ExecutionModeArgs): Promise<TransactionExecutionMode> => {
+    // TODO should be
```
Comment (krzysu) [inline packages/react/src/transactions/publications/useExecutionMode.ts:46]: anything left to do here?

**Review by krzysu — APPROVED** [review]
looks good, changeset is missing

File: packages/react/src/transactions/adapters/publications/CreateOnChainMirrorGateway.ts (hunk 2 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/adapters/publications/CreateOnChainMirrorGateway.ts:148]: This is unfortunately the field name.

File: packages/react/src/transactions/publications/useExecutionMode.ts (hunk 3 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/publications/useExecutionMode.ts:46]: nope

**Pull request merged**